Adds caching, year selection, optional forcing for shapefile downloads

spdustin opened the pull request:
- Adds `MAPS_CACHE_FOLDER` (will be created if it doesn't exist)
- Adds `MAPS_SOURCE_YEAR` to allow easier updating of source URLs for shapefiles
- Adds `REQUESTS_TIMEOUT` for census.gov's terrible server
- Clarifies processing for shapely.Linestrings during import of geometry, and guards against unexpected geometry types
- Adds prompt when `make maps` is run to force downloads to rebuild Shapely's base geometry files
- Finally, updates base geometry files to use 2024 data

jtfedd commented:
Overall looks great! Thanks for updating the maps and tool! There are some failing lints on the workflow but I think the test workflow failure is unrelated, I will look into that.

spdustin commented:
I'm happy to correct those lint errors, if you want me to update this commit

spdustin commented:
I corrected the lint errors, and added typing and a guard to the `drawSegs()` function. Given that the resulting BAM files didn't change, I'm calling that a successful test of the guard, which checks that each coordinate in a collection is 2 dimensions.

jtfedd commented:
Looks great! I refactored the test reporter so it should work correctly with forks now. I also went ahead and relaxed the lint on snake case vs camel case, I don't really mind either way and that's probably a bit overly strict. I pushed up one quick formatting fix, `make format` _should_ format everything exactly the way CI wants it. Going to go ahead and merge, thanks again for the improvements to this tool!

spdustin commented:
My pleasure, @jtfedd. Happy to contribute. I've been working on a local fork to add some other convenience features. Is there anything in particular you need help with?
